videoproviders/api/videofront.py

```diff
@@ -168,6 +168,23 @@ def get_video_with_subtitles(self, video_id):
         """Return a single video, identified by its id, along with the
         subtitles information. This method is used by the video xblock to fetch
         video + subtitles info in one go."""
+        # Try getting the video document directly from the S3 bucket metadata file
+        try:
+            response = requests.get(
+                "{cdn:s}/videos/{id:s}/metadata.json".format(
+                    cdn=getattr(settings, "VIDEOFRONT_CDN_BASE_URL"), id=video_id
+                )
+            )
+        except requests.exceptions.RequestException:
+            logger.error(
+                'VideoFront video "%s" could not be loaded from metadata file.',
+                video_id,
+            )
+        else:
+            if response.status_code == 200:
+                return response.json()
```

Comment:
I'm wondering if you should handle the case when the json file is invalid (`requests.exceptions.JSONDecodeError`), to log an error and fallback the default behavior.

Reply:
Good idea I will. As long as we log the error and can go handle it.
In theory it should not happen as I crafted each file from a script and everything is now readonly in VideoFront.

Reply:
@madmatah done. I found other cases of failures (it's such an old version of requests! :man_shrugging:)